### PR TITLE
feat: expose index cache configure via open dataset API

### PR DESCRIPTION
**Pull request by eddyxu**

### CODE REVIEW

File: python/python/lance/__init__.py [hunk 1]
```diff
@@ -77,6 +78,10 @@ def dataset(
         If specified, use the provided commit handler to lock the table while
         committing a new version. Not necessary on object stores other than S3
         or when there are no concurrent writers.
+    index_cache_size : optional, int
+        Index cache size. Index cache is a LRU cache with TTL. This number specify the
+        number of index page, for example, an IVF partition, to be cached in
```
Comment (changhiskhan) [inline python/python/lance/__init__.py:83]: This is for indexing OOM? Could you provide an example for users to estimate the max amount of memory used for the cache given parameter values? As it stands it's unclear how to translate from index pages to GBs

Comment (eddyxu) [reply python/python/lance/__init__.py:83]: Added more explains.

File: python/python/lance/__init__.py [hunk 2]
```diff
@@ -77,6 +78,15 @@ def dataset(
         If specified, use the provided commit handler to lock the table while
         committing a new version. Not necessary on object stores other than S3
         or when there are no concurrent writers.
+    index_cache_size : optional, int
+        Index cache size. Index cache is a LRU cache with TTL. This number specifies the
+        number of index page, for example, an IVF partition, to be cached in
```
Comment (westonpace) [inline python/python/lance/__init__.py:83]:
```suggestion
        number of index pages, for example, IVF partitions, to be cached in
```

File: python/python/lance/__init__.py [hunk 3]
```diff
@@ -77,6 +78,15 @@ def dataset(
...
+        the host memory. Default value is ``256``.
+
+        Roughly, for an ``IVF_PQ`` partition with ``n`` rows, the size of each index
+        page equals the combination of the pq code (``nd.array([n,pq], dtype=uint8))``
+        and the row ids (``nd.array([n], dtype=uint64)``).
+        Approximately, ``n = Total Rows / number of IVF``.
```
Comment (westonpace) [inline python/python/lance/__init__.py:89]:
```suggestion
        Approximately, ``n = Total Rows / number of IVF partitions``.
        ``pq = number of PQ sub-vectors``.
```

**Review by westonpace — COMMENTED** [review]
We could probably write an entire page of documentation on the index cache, how it works, and recommendations, but this probably isn't the spot for it.

**Review by westonpace — APPROVED** [review]

**Pull request merged**